### ISSUE TITLE
Enable clear the UndoManager and callbacks in undo()/redo() actions

### ISSUE THREAD

**Issue by chemitaxis** [mobxjs/mobx-state-tree#1107]
**_Feature request_**
Enable clear the UndoManager and callbacks in undo()/redo() actions

**Is your feature request related to a problem? Please describe.**
It's not a problem, it's a feature to implement.

**Describe the solution you'd like**
I have implemented

**Describe alternatives you've considered**
Overwrite the UndoManager.

**Additional context**

Sometimes you would need to clear the UndoManager, for example, an error. Sometimes you need to execute a custom callback function after you can redo()/undo() actions

**Are you willing to (attempt) a PR?**
* [X ] Yes
* [ ] No

**Comment by mweststrate** [comment]
The `clear` is clear enough :). 

About the callbacks, are there any specific benefits of doing `undomanager.undo(()=> somethingElse())` over `undoManager.undo(); somethingElse();`?

**Comment by chemitaxis** [comment]
Hi @mweststrate I'm going to answer with another question... (sorry 😄 ). If a run the code on this way:

```
undoManager.undo();
somethingElse();
```

Is this code to run in a synchronous way? If the answer is yes I can remove the part of the callbacks stuff. But (maybe I'm wrong) I remember a case when the DOM was updated and we needed to execute the callback inside the undo method because the re-render of the components was later and we had conflicts with it. Thanks.
